torch/nn/modules/module.py

```diff
@@ -1089,6 +1095,17 @@ def _call_impl(self, *input, **kwargs):
         if bw_hook:
             result = bw_hook.setup_output_hook(result)
 
+        if self._pre_backward_hooks:
+            var = result
+            while not isinstance(var, torch.Tensor):
```

Comment:
This is not the right way to do backward hook. This might trigger at the wrong time and not contain all gradients. Like the old backward hooks.
You want to do something closer to the full backward hooks above.